src/main/java/co/cask/hydrator/plugin/batchSource/KafkaBatchSource.java

```diff
@@ -304,7 +304,10 @@ public Schema getMessageSchema() {
     }
 
     public void validate() {
-      getBrokerMap();
+      // brokers can be null since it is macro enabled.
+      if (kafkaBrokers != null) {
+        getBrokerMap();
```

Comment:
if someone uses a macro for brokers, and provides an invalid value at runtime, will a proper error be thrown?